src/Components/Modals/AddAdditionalRepositoriesModal.js

```diff
@@ -95,19 +97,57 @@ const AddAdditionalRepositoriesModal = (props) => {
           error={additionalRepositoriesError}
           selectedRepositories={selectedRepositories}
           setSelectedRepositories={setSelectedRepositories}
+          isSubmitting={isSubmitting}
         />
       </Modal>
     </React.Fragment>
   );
 };
 
+const SubmitButton = ({
+  selectedRepositories,
+  submitForm,
+  isSubmitting,
+  isLoading: additionalRepositoriesAreLoading,
+}) => {
+  const [buttonState, setButtonState] = useState('notClicked');
+  const handleClick = () => {
+    setButtonState('clicked');
+    submitForm();
+  };
+
+  return (
+    <ActionGroup>
+      <Button
+        variant="primary"
+        onClick={handleClick}
+        isLoading={additionalRepositoriesAreLoading}
+        isDisabled={isSubmitting || selectedRepositories.length === 0}
+        spinnerAriaValueText="Saving Changes..."
+      >
+        {buttonState === 'notClicked' && 'Save Changes'}
+        {buttonState === 'clicked' && 'Saving Changes'}
```

Comment:
These can also both be `isSubmitting` right? Then we don't even need the `buttonState`

Reply:
Cant remove the button state to isSubmitting , the button will not change appropriately after click.